storage/innobase/fil/fil0crypt.cc

```diff
@@ -1441,9 +1441,15 @@ space_list_t::iterator fil_space_t::next(space_list_t::iterator space,
     fil_space_t *next_space=
       fil_system.default_encrypt_next(space == fil_system.space_list.end()
 				      ? nullptr : &*space, recheck, encrypt);
-    space= next_space
-      ? space_list_t::iterator(next_space)
-      : fil_system.space_list.end();
+
+	if (next_space == nullptr || next_space == fil_system.temp_space)
+	{
+		space = fil_system.space_list.end();
+	}
+	else
+	{
+		space = space_list_t::iterator(next_space);
+	}
```

Comment:
Is there any reason why `fil_system.temp_space` is part of `fil_system.space_list` in the first place? I can’t think of any. Rather than complicating the logic, it would be better to add `ut_ad(space != fil_system.temp_space)` and fix some other code so that it holds.